pyspecdata/core.py

```diff
@@ -1616,109 +1622,134 @@ def concat(datalist, dimname, chop=False):
             )
         )
     other_info_out = datalist[0].other_info
-    for j in range(0, len(datalist)):
-        # {{{ make list for the shape to check, which contains the dimensions we are NOT concatting along
-        if dimname in shapes[j].dimlabels:
-            newdimsize += shapes[j][dimname]
-            shapetocheck = list(shapes[j].shape)
-            shapetocheck.pop(shapes[j].axn(dimname))
-        else:
+    if dimname in datalist[0].dimlabels:
+        print(
+            "The dimname you specified exists already in one of the datasets so I am assuming it also exists in the other"
+        )
+        # {{{ make list of the data axes and dimname axes for datasets
+        ndarrays = []  # will contain numpy array of data for each dataset
+        conc_axis = [] # will contain numpy array of the dimname axis for each dataset
+        for j in range(len(datalist)):
+            ndarrays.append(datalist[j].data)
+            conc_axis.append(datalist[j][dimname])
+        # }}}
+        # {{{ concatenate the data ndarrays and dimname axis ndarrays
+        newdata = np.concatenate(
+            tuple(ndarrays), axis=datalist[-1].dimlabels.index(dimname)
+        )
+        new_dimname = np.concatenate(tuple(conc_axis))
+        # }}}
+        # {{{ Pull axis labels from the last item in the list
+        #     note this is similar to what is done below if the dimname
+        #     is not preexisting however here there is no newdimsize or
+        #     dimname so there is a slight difference
+        extra_dims = False
+        if len(datalist[-1].axis_coords) > 1:
+            extra_dims = True
+            extra_dimlabels = list(datalist[-1].dimlabels)
+            extra_axis_coords = list(datalist[-1].axis_coords)
+            dimname_idx = extra_dimlabels.index(dimname)
+            extra_dimlabels.pop(dimname_idx)
+            extra_axis_coords.pop(dimname_idx)
+        # }}}
```

Comment:
inspired by lines 1731 below but I add up here since we don't want to include the newdimsize stuff that is done down there